### DISCUSSION TITLE
enqueueActions not allowed in setup function

### DISCUSSION THREAD

**Discussion by GalStan** [statelyai/xstate#4719] [Ideas]
During migrating to version 5 I found that it's not possible to substitute `pure` actions with `enqueueActions` in action declarations in `setup` function due to difference in expecting typings. Is it bug? Or is there any reason to implement it like this?

**Comment by Andarist** [comment]
It's currently not possible because we are hitting TS inference limits. You might want to follow this effort that is meant to alleviate the problem: https://github.com/statelyai/xstate/pull/4703
